src/incidencehandler.cpp

```diff
@@ -367,37 +393,41 @@ KCalCore::Incidence::Ptr IncidenceHandler::incidenceToExport(KCalCore::Incidence
         if (sendWithoutDtEnd) {
             // A single-day all-day event was received without a DTEND, and it is still a single-day
             // all-day event, so remove the DTEND before upsyncing.
-            LOG_DEBUG("Remove DTEND from" << incidence->uid());
+            LOG_DEBUG("Removing DTEND from" << incidence->uid());
             event->setDtEnd(KDateTime());
-        } else if (event->hasEndDate()) {
-            KDateTime dt = event->dtEnd();
-            // Event::dtEnd() is inclusive, but DTEND in iCalendar format is exclusive.
-            LOG_DEBUG("Adding +1 day to" << dt.toString() << "to make exclusive DTEND" << "for" << incidence->uid());
-            dt = dt.addDays(1);
-            event->setDtEnd(dt);
         } else {
-            KDateTime dt = event->dtStart().addDays(1);
-            LOG_DEBUG("Adding DTEND of DTSTART+1" << dt.toString() << "for" << incidence->uid());
+            KDateTime dt;
+            if (event->hasEndDate()) {
+                // Event::dtEnd() is inclusive, but DTEND in iCalendar format is exclusive.
+                dt = KDateTime(event->dtEnd().addDays(1).date(), event->dtEnd().timeSpec());
```

Comment:
I'm not sure about this.  Am I allowed to unconditionally strip time information from all-day events or are all-day events with time information supported by CalDAV / iCal?